luigi/tasks/genome_mapping/genome_mapping_tasks.py

```diff
@@ -0,0 +1,395 @@
+            if region['name'] == sequence_name:
+                return region['coord_system']
+        return 'chromosome'
+
+    def get_assembly_name(self, ensembl_json):
+        """
+        Get assembly name that is used in the FTP filename.
+        """
+        if self.species in ['arabidopsis_lyrata', 'phaeodactylum_tricornutum']:
+            return ensembl_json['assembly_name']
+        else:
+            return ensembl_json['default_coord_system_version']
+
+    def output(self):
+        if not GetAssemblyInfoJson(species=self.species, division=self.division).complete():
+            GetAssemblyInfoJson(species=self.species, division=self.division).run()
+        ensembl_json = GetAssemblyInfoJson(species=self.species, division=self.division).output().path
+        data = json.load(open(ensembl_json, 'r'))
+        chromosomes = []
+        for region in data['top_level_region']:
+            if region['name'] not in data['karyotype']:
+                chromosomes.append('{species}.{assembly}.dna.nonchromosomal.fa'.format(
+                                    species=self.species.capitalize(),
+                                    assembly=self.get_assembly_name(data)))
+                break
+        for chromosome in data['karyotype']:
+            filename = '{species}.{assembly}.dna.{sequence_type}.{chromosome}.fa'.format(
+                species=self.species.capitalize(),
+                assembly=self.get_assembly_name(data),
+                chromosome=chromosome,
+                sequence_type=self.get_sequence_type(data, chromosome)
+            )
+            chromosomes.append(filename)
+        return chromosomes
+
+
+class DownloadChromosome(luigi.Task):
+    """
+    Download and uncompress chromosome fasta file.
+    """
+    species = luigi.Parameter()
+    chromosome = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def output(self):
+        genome_path = genome_mapping().genome(self.species)
+        return luigi.LocalTarget(os.path.join(genome_path, self.chromosome))
+
+    def get_url(self):
+        url = None
+        if self.division == 'Ensembl':
+            url = 'ftp://ftp.ensembl.org/pub/current_fasta/{species}/dna/{chromosome}.gz'
+        elif self.division == 'EnsemblFungi':
+            url = 'ftp://ftp.ensemblgenomes.org/pub/current/fungi/fasta/{species}/dna/{chromosome}.gz'
+        elif self.division == 'EnsemblMetazoa':
+            url = 'ftp://ftp.ensemblgenomes.org/pub/current/metazoa/fasta/{species}/dna/{chromosome}.gz'
+        elif self.division == 'EnsemblPlants':
+            url = 'ftp://ftp.ensemblgenomes.org/pub/current/plants/fasta/{species}/dna/{chromosome}.gz'
+        elif self.division == 'EnsemblProtists':
+            url = 'ftp://ftp.ensemblgenomes.org/pub/current/protists/fasta/{species}/dna/{chromosome}.gz'
+        url = url.format(species=self.species, chromosome=self.chromosome)
+        return url
+
+    def run(self):
```

Comment:
For this we should we use the fact that we are on the same file system to copy the files over. That should be quicker. Additionally, I have a `tasks.utils.fetch.FetchTask` helper that will either copy a file (given a path), or fetch something via FTP (given an FTP url). Doesn't unzip though. Plus it looks like all files are in the same directory so we could just copy the whole directory at once. That would be easy.

Reply:
I like keeping the pipeline independent of the EBI cluster. As implemented, the genome download can be run anywhere (like Rfam in the cloud etc). Will it ever happen? I don't know, but FTP doesn't seem to slow things down so I don't see many advantages of tying us up to the filesystem